backend/cmm_helpers.ml

```diff
@@ -3618,8 +3618,79 @@ let sequence x y =
   | _, Ctuple [] -> x
   | _, _ -> Csequence (x, y)
 
+let rec is_small_and_pure_int_expr ~size_bias ~size ~toplevel expr =
+  if size > 12 - size_bias
+  then None
+  else
+    match expr with
+    | Cconst_int _ | Cconst_natint _ | Cconst_symbol _ -> Some (size + 1)
```

Comment:
(Not completely sure about `Cconst_symbol`,
or maybe it should be considered bigger.)